lib/chef/http/authenticator.rb

```diff
@@ -104,6 +152,51 @@ def load_signing_key(key_file, raw_key = nil)
         raise Chef::Exceptions::InvalidPrivateKey, msg
       end
 
+      def self.retrieve_certificate_key(client_name)
+        require "openssl" unless defined?(OpenSSL)
+
+        if ChefUtils.windows?
+          # This code block assumes a certificate with a subject name like "Chef-<node-name>" is in the \LocalMachine\My store and
+          # that there is a password stored in the registry to be used to export the pfx with.
+          powershell_password_code = <<~CODE
+              Try {
+                  Get-ItemPropertyValue -Path "HKLM:\\Software\\Progress\\Authenticator" -Name "PfxPass" -ErrorAction Stop;
+              }
+              Catch {
+                  return $false
+              }
+          CODE
+          password = powershell_exec!(powershell_password_code).result
+
+          if password == false
+            return false
+          end
+
+          powershell_code = <<~CODE
+              Try {
+                  $my_pwd = ConvertTo-SecureString -String "#{password}" -Force -AsPlainText;
+                  $cert = Get-ChildItem -path cert:\\LocalMachine\\My -Recurse | Where-Object { $_.Subject -match "#{client_name}" } -ErrorAction Stop;
+                  $tempfile = [System.IO.Path]::GetTempPath() + "export_pfx.pfx";
+                  Export-PfxCertificate -Cert $cert -Password $my_pwd -FilePath $tempfile;
```

Comment:
Let's also look and see if chef/win32-certstore meets the needs, might be able to get rid of a lot of the PS bits.

Reply:
Had to punt this for now. In doing some exploration work to replace that code with Ruby code, I found that I would need at least a whole sprint and probably longer to replace add the C++/FFI code to Win32-Certstore to make this all work. I created a ticket for that effort and put all my notes and code examples I had tested with in it.

Reply:
I took a look at this and I can't see an easy way of splicing in the win32-certstore code without another side project - the issue is that I need more time than is in the sprint to get the C++ code in shape to export a pfx. It's not too complicated but will require 4-5 days to get it working nicely. Added a ticket for this and falling back to the PowerShell code above for now.